sycamore/transforms/embed.py

```diff
@@ -85,7 +92,69 @@ def generate_embeddings(self, doc_batch: list[Document]) -> list[Document]:
         text_batch = ["" if doc.text_representation is None else doc.text_representation for doc in doc_batch]
         embeddings = self._transformer.encode(text_batch, batch_size=self.model_batch_size, device=self.device)
         for doc, embedding in zip(doc_batch, embeddings):
-            doc.embedding = embedding
+            doc.embedding = embedding.tolist()
+
+        return doc_batch
+
+
+class OpenAIEmbeddingModels(Enum):
+    TEXT_EMBEDDING_ADA_002 = "text-embedding-ada-002"
+
+
+class OpenAIEmbedder(Embedder):
+    """Embedder implementation using the OpenAI embedding API.
+
+    Args:
+        model_name: The name of the OpenAI embedding model to use.
+        batch_size: The Ray batch size.
+        model_batch_size: The number of documents to send in a single OpenAI request.
+    """
+
+    def __init__(
+        self,
+        model_name: str = OpenAIEmbeddingModels.TEXT_EMBEDDING_ADA_002.value,
+        batch_size: Optional[int] = None,
+        model_batch_size: int = 100,
+        api_key: Optional[str] = None,
+    ):
+        super().__init__(model_name, batch_size, model_batch_size, device="cpu")
+
+        if api_key is None:
+            api_key = os.environ.get("OPENAI_API_KEY", None)
+
+        assert api_key is not None, (
+            "You must provide an API key to "
+            "use the LLM. Either pass it in "
+            "the constructor or set the "
+            "OPENAI_API_KEY environment "
+            "variable."
+        )
+        self.api_key = api_key
+        self.model_name = model_name
+
+    @retry(
+        stop=stop_after_attempt(3),
+        wait=wait_random(min=1, max=2),
+        retry=retry_if_exception_type(openai.error.RateLimitError),
+    )
+    def _openai_embeddings(self, text_to_embed: list[str]) -> list:
+        return openai.Embedding.create(input=text_to_embed, engine=self.model_name).data
+
+    def generate_embeddings(self, doc_batch: list[Document]) -> list[Document]:
+        # TODO: Add some input validation here.
+        # The OpenAI docs are quite vague on acceptable values for model_batch_size.
+
+        for batch in batched(doc_batch, self.model_batch_size):
+            text_to_embed = [
+                doc.text_representation.replace("\n", " ") for doc in batch if doc.text_representation is not None
```

Comment:
why we need "\n" get replaced?

Reply:
It was recommended in the OpenAI libraries (https://github.com/openai/openai-python/blob/284c1799070c723c6a553337134148a7ab088dd8/openai/embeddings_utils.py#L20).